web/src/LogStore.ts

```diff
@@ -106,13 +117,28 @@ class LogStore {
     let startIndex = 0
     let lastIndex = this.segments.length - 1
     if (filteredLogs) {
-      let span = this.spans[mn]
-      if (!span) {
+      let spans = this.spansInManifest(mn)
+      let earliestStartIndex = -1
+      let latestEndIndex = -1
+      for (let spanId in spans) {
+        let span = spans[spanId]
+        if (
+          earliestStartIndex == -1 ||
+          span.firstSegmentIndex < earliestStartIndex
+        ) {
+          earliestStartIndex = span.firstSegmentIndex
+        }
+        if (latestEndIndex == -1 || span.lastSegmentIndex > latestEndIndex) {
+          latestEndIndex = span.lastSegmentIndex
+        }
+      }
+
+      if (earliestStartIndex == -1) {
         return ""
       }
 
-      startIndex = span.firstSegmentIndex
-      lastIndex = span.lastSegmentIndex
+      startIndex = earliestStartIndex
+      lastIndex = latestEndIndex
```

Comment:
/me resists make a snide web assembly comment

Reply:
?

Reply:
if the comment is "why don't we compile the go code to webassembly"...i think we _might_ go down that path eventually, but it's too early to say. these two data structures might end up diverging a lot because the needs of the UI are going to be different...

Reply:
Oh it was a joke, sorry!